src/Umbraco.Web.UI.Client/src/views/propertyeditors/blockgrid/prevalue/blockgrid.stylesheetpicker.controller.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * @ngdoc controller
+ * @name Umbraco.Editors.BlockGrid.BlockConfigurationController
+ * @function
+ *
+ * @description
+ * The controller for the content type editor property settings dialog
+ */
+
+(function () {
+    "use strict";
+
+    function StylesheetPickerController($scope, localizationService, editorService, overlayService) {
+
+        //var unsubscribe = [];
+
+        var vm = this;
+
+        vm.addStylesheet = function() {
+            localizationService.localize("blockEditor_headlineAddCustomStylesheet").then(localizedTitle => {
+  
+                const filePicker = {
+                    title: localizedTitle,
+                    isDialog: true,
+                    filter: i => {
+                        return !(i.name.indexOf(".css") !== -1);
+                    },
+                    filterCssClass: "not-allowed",
+                    select: node => {
+                        const filepath = decodeURIComponent(node.id.replace(/\+/g, " "));
+                        $scope.model.value = "~/" + filepath.replace("wwwroot/", "");
+                        editorService.close();
+                    },
+                    close: () => editorService.close()
+                };
+
+                editorService.staticFilePicker(filePicker);
```

Comment:
Isn't this what we already have here?
https://github.com/umbraco/Umbraco-CMS/blob/78786e942fa302b011ed9e4782a28d1f32f6f585/src/Umbraco.Web.UI.Client/src/views/propertyeditors/blocklist/prevalue/blocklist.blockconfiguration.overlay.controller.js#L226

Could this be added as an infinite editor in `editorService`.
E.g. `editorService.styleSheetPicker`?